VSCode ext: `ccdrec` snippet uses `mut` host without `mutable` attribute

**Bug Description**

In the VSCode extension, the `ccdrec` snippet uses `mut` host without `mutable` attribute, which causes a compiler error.

**Steps to Reproduce**
1. Use the snippet `ccdrec` and enter values of existing data types and contracts in the placeholders.
2. Return a value from the function.
3. Compile

**Expected Result**
Compiles successfully.

**Actual Result**
A compiler error because the snippet uses `&mut impl HasHost...` instead of `&impl HasHost` when the `mutable` attribute isn't specified.
```
error[E0308]: mismatched types
   --> src/lib.rs:105:1
    |
105 | / #[receive(
106 | |     contract = "vstest",
107 | |     name = "foo",
108 | |     parameter = "()",
109 | |     return_value = "()",
110 | |     error = "()"
111 | | )]
    | |__^ types differ in mutability
112 |   fn vstest_foo<S: HasStateApi>(
    |      ---------- arguments to this function are incorrect
    |
    = note: expected mutable reference `&mut _`
                       found reference `&ExternHost<_>`
note: function defined here
   --> src/lib.rs:112:4
    |
112 | fn vstest_foo<S: HasStateApi>(
    |    ^^^^^^^^^^
113 |     ctx: &impl HasReceiveContext,
114 |     host: &mut impl HasHost<State, StateApiType = S>
    |     ------------------------------------------------
    = note: this error originates in the attribute macro `receive` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Versions**
- Software Version: 0.1.0-0
- OS: macOS Ventura (Apple M2 chip)

--- a/vscode-smart-contracts/README.md
+++ b/vscode-smart-contracts/README.md
@@ -32,12 +32,12 @@ Searches the workspace for common smart contract tasks.
 
 ### Provide snippets for smart contract development
 
-A number of snippets are provided, all prefixed with `ccd` (for smart contract).
+A number of snippets are provided, all prefixed with `ccd` (for Concordium).
 Just type `ccd` in your smart contract source file and let IntelliSense show you the list of snippets.
 
 ## Requirements
 
-The extension relies on [`rustup`](https://rustup.rs/) and `cargo` being installed and available in PATH.
+The extension relies on [`rustup`](https://rustup.rs/) and [`cargo`](https://doc.rust-lang.org/cargo/) being installed and available in PATH.
 
 ## Extension Settings
 

--- a/vscode-smart-contracts/snippets.json
+++ b/vscode-smart-contracts/snippets.json
@@ -11,7 +11,7 @@
       ")]",
       "fn ${1/[\\.-]/_/g}_${2/[\\.-]/_/g}<S: HasStateApi>(",
       "\tctx: &impl HasReceiveContext,",
-      "\thost: &mut impl HasHost<State<S>, StateApiType = S>",
+      "\thost: &impl HasHost<State<S>, StateApiType = S>",
       ") -> Result<$4, $5> {",
       "\tlet parameter: $3 = ctx.parameter_cursor().get()?;",
       "\t$0",

--- a/vscode-smart-contracts/src/cargo-concordium.ts
+++ b/vscode-smart-contracts/src/cargo-concordium.ts
@@ -89,7 +89,7 @@ export async function build(
     taskDefinition,
     scope,
     "Build smart contract",
-    cwd,
+    cwd === "." ? "concordium" : cwd,
     new vscode.ProcessExecution(executable, ["concordium", "build", ...args], {
       cwd,
     })
@@ -113,7 +113,7 @@ export async function test(
     taskDefinition,
     scope,
     "Test smart contract",
-    cwd,
+    cwd === "." ? "concordium" : cwd,
     new vscode.ProcessExecution(executable, ["concordium", "test", ...args], {
       cwd,
     })

--- a/vscode-smart-contracts/src/extension.ts
+++ b/vscode-smart-contracts/src/extension.ts
@@ -72,6 +72,7 @@ const taskProvider: vscode.TaskProvider = {
               defaultOutDir,
               "--schema-base64-out",
               path.join(defaultOutDir, "module-schema.bs64"),
+              "--schema-embed",
             ];
             return [
               cargoConcordium.build(cwd, workspaceFolder, defaultArgs),
@@ -125,7 +126,7 @@ const taskProvider: vscode.TaskProvider = {
       const workspaceFolder = vscode.workspace.getWorkspaceFolder(
         vscode.Uri.file(cwd)
       );
-      const resolvedTask = await cargoConcordium.build(
+      const resolvedTask = await cargoConcordium.test(
         cwd,
         workspaceFolder,
         definition.args ?? []

